### PR TITLE
feat: allow model selection in client

### PR DESCRIPTION
**Pull request by ZiniuYu**
This pr allows the user to select the model in the client when multiple models are running.

Sample usage:

```
from clip_client import Client

c = Client(xxx)
c.encode(xxx, parameters={'model': 'RN50'})
c.rank(xxx, parameters={'model': 'RN50'})
```

### CODE REVIEW

File: docs/user-guides/client.md [hunk 1]
```diff
@@ -192,6 +192,65 @@ Use `.encode(..., show_progress=True)` to turn on the progress bar.
...
+  - name: ViT_B_32
+    replicas: 1
+    uses:
+      jtype: CLIPEncoder
+      with:
+        name: ViT-B/32
+      metas:
+        py_modules:
+          - clip_server.executors.clip_torch
+      requests: {'/encode/ViT-B/32': 'encode', '/': '_dry_run_func'}
```
Comment (numb3r3) [inline docs/user-guides/client.md:220]: Let's first hide this feature. This is a very advanced usage.

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: client/clip_client/client.py [hunk 2]
```diff
@@ -185,8 +189,11 @@ def _iter_doc(self, content) -> Generator['Document', None, None]:
                 )
 
     def _get_post_payload(self, content, kwargs):
+        parameters = kwargs.get('parameters', {})
+        target_model = parameters.get('model', None)
+        endpoint = f'/encode{f"/{target_model}" if target_model else ""}'
```
Comment (numb3r3) [inline client/clip_client/client.py:194]:
```suggestion
        model_name = parameters.get('model', '')
        endpoint = f'/encode/{model_name}'
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: client/clip_client/client.py [hunk 3]
```diff
@@ -185,8 +189,11 @@ def _iter_doc(self, content) -> Generator['Document', None, None]:
...
         payload = dict(
-            on='/',
+            on=endpoint,
```
Comment (numb3r3) [inline client/clip_client/client.py:196]:
```suggestion
            on=f'/encode/{model_name}',
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: client/clip_client/client.py (hunk 2 above)
Comment (ZiniuYu) [reply client/clip_client/client.py:194]: smart

File: client/clip_client/client.py (hunk 3 above)
Comment (ZiniuYu) [reply client/clip_client/client.py:196]: are we introducing extra '/'?

**Review by numb3r3 — APPROVED** [review]

**Pull request merged**